src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs

```diff
@@ -18,9 +18,16 @@ public abstract class BaseConverterOneWay<TFrom, TTo> : BaseConverterOneWay
 	/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
 	/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
 	/// <returns>An object of type <typeparamref name="TTo"/>.</returns>
-	public override object? Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
+	public override sealed object? Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
 	{
-		if (value is not TFrom valueFrom)
+		if (value is null && IsNullable<TFrom>())
+		{
+#pragma warning disable CS8604 // Possible null reference argument.
```

Comment:
I don't understand the need for this. I expect that if `TFrom` is not nullable then default would not return `null`. Take a `bool` for example, `default(bool)` is `false` not `null` whereas `default(bool?)` would be `null`.

Do you know if this is needed and why?

Reply:
Yea, it's a bit of a weird case and it took me a minute to understand it as well. 

In `if (value is not TFrom valueFrom)`, `valueFrom` will never be `null`.

So if a user passes in `null` to the converter, `if (value is not TFrom valueFrom)` will resolve to `true` and we'll throw an `ArgumentException`.

You can see this by removing the `if (value is null && IsNullable<TFrom>())` block and running the unit tests.

Reply:
Hmm ok, I think I might be getting it, `TFrom` is null and `ConvertFrom` doesn't specify that `TFrom` can be null (we require the inheritor declares that which we check inside `IsNullable<>`.

But wait a minute why is `ConvertFrom(default(TFrom)` not valid as it's claiming `default(TFrom)` could be null but if it could be null surely that means the signature of `ConvertFrom(TFrom ...)` would be allowing for `TFrom` to be null? Now I am even more confused...

Although on this topic given the above does that mean we could change the signature of `ConvertFrom` from:

```csharp
public abstract TTo? ConvertFrom(TFrom value);
```

to:

```csharp
public abstract TTo ConvertFrom(TFrom value);
```

This would then let the inheriting converter implementation determine whether a null can be returned?

Reply:
The reason I use `return ConvertFrom(default)` instead of `return ConvertFrom(null)` or `return ConvertFrom((TFrom)null)` is mostly thanks to the Roslyn analyzers.

In this case, since we're confident that `value` is `null` and that `TFrom` is nullable, return `default` is the same as returning `null`.

Roslyn gets mad if we try to do something like `return `ConvertFrom((TFrom)null)`, because it doesn't know whether or not `TFrom` is nullable.

Reply:
OK that makes some sense, so I guess in an ideal world we could inform Roslyn that `TFrom` is in fact nullable through our `IsNullable` method call? e.g. using something like `[NullWhen]` or similar. I don't know if there is something that fits the bill here though.

Reply:
Right! But there's no support for generics in the [Nullable Static Analyzers](https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/nullable-analysis), so there's no way for us to tell Roslyn that TFrom is nullable and/or that `value` is null. 

And you're correct that the class that inherits/implements `BaseConverterOneWay<TFrom, TTo>` chooses whether or not `TFrom` and `TTo` are nullable.

So we're kinda caught in this weird in-between place where this is the best solution we can do until a new version of C#/Roslyn helps us out.

Reply:
OK. Thanks for helping my understanding here 😄